pkg/harvester/utils/feature-flags.js

```diff
@@ -14,16 +14,35 @@ export function getVersion(v) {
   }
 
   try {
+    // v1.4.1-rc.1 => v1.4.1, v1.4.1-dev-20241222 => v1.4.1
     return `v${ semver.major(v) }.${ semver.minor(v) }.${ semver.patch(v) }`;
   } catch (error) {
     // fallback to the latest version
-    return Object.keys(RELEASE_FEATURES).sort((a, b) => semver.compare(a, b)).pop();
+    return latestVersion(Object.keys(RELEASE_FEATURES));
   }
 }
 
+function latestVersion(versions) {
+  return versions.sort((a, b) => semver.compare(a, b)).pop();
```

Comment:
[Nit] we can use `semver.maxSatisfying` to find the highest matching version directly.
e.g. `semver.maxSatisfying(versions, '*');`
Ref: https://github.com/npm/node-semver?tab=readme-ov-file#ranges-1